src/components/DateRangePicker.jsx

```diff
@@ -263,32 +265,36 @@ export default class DateRangePicker extends React.Component {
       onFocusChange,
     } = this.props;
 
+    const onOutsideClick = !withPortal && !withFullScreenPortal ? this.onOutsideClick : () => {};
```

Comment:
It would be more performant to cache this noop function outside the render path, so react can know when it's the same object. Alternatively, set it to null/undefined, and it won't be passed at all?